### PR TITLE
fix for microsoft store failing to create virtual environment due to …

### PR DESCRIPTION
**Pull request by bschnurr**
…outdated virtualenv

- also fix for when env folder is missing a "lib" folder (could be due to improper env path)
- fix for repl rootpath (which is null), when iterating over workspace contexts

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs [hunk 1]
```diff
@@ -162,6 +190,13 @@ private static async Task ContinueCreate(IServiceProvider provider, IPythonInter
                 }
             }
 
+            try {
+                if (!useVEnv) {
```
Comment (rchiodo) [inline Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs:194]: I think we might need to ask the user here? We're running something on their behalf in their global python.

File: Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs [hunk 2]
```diff
@@ -148,9 +148,37 @@ public static async Task<IPythonInterpreterFactory> CreateAndAddFactory(
...
+                        output.Show();
+                    }
+                }
+            }
+        }
+
         private static async Task ContinueCreate(IServiceProvider provider, IPythonInterpreterFactory factory, string path, bool useVEnv, Redirector output) {
             path = PathUtils.TrimEndSeparator(path);
-            var name = Path.GetFileName(path);
+            var name =Path.GetFileName(path);
```
Comment (AdamYoblick) [inline Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs:181]: nit: space after the equals

**Review by AdamYoblick — APPROVED** [review]

File: Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs (hunk 1 above)
Comment (bschnurr) [reply Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs:194]: ah true. i think there is an example of this.. i'll try to find where we do something similar

Comment (bschnurr) [reply Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs:194]: I think i'll try to detect this error case instead and just print this command for the user to run themselves

**Review by rchiodo — APPROVED** [review]

**Pull request merged**